src/apps/js_generic/js_generic.cpp

```diff
@@ -813,9 +814,22 @@ namespace ccfapp
         JS_NewCFunction(ctx, ccfapp::js_wrap_key, "wrapKey", 3));
 
       auto kv = JS_NewObjectClass(ctx, kv_class_id);
-      JS_SetOpaque(kv, &args.tx);
+      JS_SetOpaque(kv, &tx);
       JS_SetPropertyStr(ctx, ccf, "kv", kv);
 
+      auto state = JS_NewObject(ctx);
+      auto kv_tx_id = tx.get_read_tx_id();
+      ccf::TxID tx_id;
+      tx_id.view = static_cast<ccf::View>(kv_tx_id.term);
+      tx_id.seqno = static_cast<ccf::SeqNo>(kv_tx_id.version);
```

Comment:
This is very annoying. I'm doing this so that I can `to_str()`.  The casts are needed as `kv::TxId` uses signed, while `ccf::TxID` uses unsigned ints.